### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by sebdesign**
Addresses #16

### CODE REVIEW

**Comment by sebdesign** [comment]
Some tests might be naive or poor, please let me know if I can do anything to improve them.

File: lib/ZlibInputStream.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ public function read(): Promise {
             }
 
             if ($data === null) {
-                $decompressed = \inflate_add($this->resource, "", \ZLIB_FINISH);
+                $decompressed = @\inflate_add($this->resource, "", \ZLIB_FINISH);
```
Comment (kelunik) [inline lib/ZlibInputStream.php]: Why did you add these error suppressions?

Comment (sebdesign) [reply lib/ZlibInputStream.php]: As per the [documentation](https://secure.php.net/manual/en/function.inflate-add.php#refsect1-function.inflate-add-errors), an `E_WARNING` error is generated if the stream is invalid or corrupt.
This allows us the `StreamException` to be raised in those cases, as well as when the function returns false without errors.

Comment (kelunik) [reply lib/ZlibInputStream.php]: But can `deflate_add` ever be invalid based on the input?

Comment (sebdesign) [reply lib/ZlibInputStream.php]: There are error suppressions in the stream functions in the resource streams also, so I thought it makes sense to have those as well.

Comment (sebdesign) [reply lib/ZlibInputStream.php]: Actually I don't think so, I couldn't find any case where the inflate/deflate would produce errors solely based on input.

Comment (sebdesign) [reply lib/ZlibInputStream.php]: My previous comment is wrong.
This test actually test that case:
`Amp\ByteStream\Test\ZlibInputStreamTest::testInvalidStream`

Comment (kelunik) [reply lib/ZlibInputStream.php]: I guess inflate might produce a warning with an invalid input.

Comment (sebdesign) [reply lib/ZlibInputStream.php]: But the error suppresssions on the `deflate_add` calls are useless. They can't produce errors based on input (I checked the unit tests in the php-source repository).

Comment (kelunik) [reply lib/ZlibInputStream.php]: A warning might also happen with `deflate_add`: https://lxr.room11.org/xref/php-src%40master/ext/zlib/zlib.c#1242

We should just ensure to include the error message in the exception message then.

Comment (bwoebi) [reply lib/ZlibInputStream.php]: It shouldn't really, except if you're doing bad things with zlib, which it doesn't here. Should be find to not error suppress calls to deflate_add.

Comment (sebdesign) [reply lib/ZlibInputStream.php]: I agree, since there are no unit tests in the php-source to confirm those cases, I believe it is safe to remove suppressions.

Comment (kelunik) [reply lib/ZlibInputStream.php]: @sebdesign Could you remove them for `deflate`?

Comment (sebdesign) [reply lib/ZlibInputStream.php]: Would you like me to change anything else?

File: test/ZlibOutputStreamTest.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    public function testInvalidOptions() {
+        $this->expectException(StreamException::class);
+
+        new ZlibOutputStream(new OutputBuffer(), \ZLIB_ENCODING_GZIP, ["level" => 42]);
+    }
+}
+
+class OutputBuffer implements OutputStream, Promise {
```
Comment (kelunik) [inline test/ZlibOutputStreamTest.php]: Please move that to `./src/OutputBuffer.php`.

File: lib/ZlibInputStream.php (hunk 1 above)
Comment (kelunik) [reply lib/ZlibInputStream.php]: I added one further comment, rest should be fine.

File: test/OutputBufferTest.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+class OutputBufferTest extends TestCase {
+    public function testWrite() {
+        Loop::run(function () {
+            $output = new OutputBuffer();
+            $output->write('foo');
+            $output->end();
+
+            $memory = new InMemoryStream(yield $output);
+
+            $this->assertSame('foo', yield $memory->read());
```
Comment (kelunik) [inline test/OutputBufferTest.php]: You can just compare `$this->assertSame('foo', yield $output);`, no need for the extra `InMemoryStream`.

**Comment by kelunik** [comment]
Thank you for your contribution!
